### PR TITLE
Event loop affinity

### PR DESCRIPTION
**Pull request by vietj**
Implementation of https://github.com/eclipse-vertx/vert.x/issues/3430

### CODE REVIEW

File: src/main/java/io/vertx/core/Vertx.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ static void clusteredVertx(VertxOptions options, Handler<AsyncResult<Vertx>> res
   /**
    * Gets the current context
    *
-   * @return The current context or null if no current context
+   * @return The current context or @code null if there is no current context
```
Comment (jponge) [inline src/main/java/io/vertx/core/Vertx.java:113]: `{@code null}`

File: src/test/java/io/vertx/core/ContextTest.java [hunk 2]
```diff
@@ -809,4 +809,10 @@ public void testWorkerExecutor() {
     });
     await();
   }
+
+  @Test
+  public void testSticky() {
```
Comment (jponge) [inline src/test/java/io/vertx/core/ContextTest.java:814]: 👍

File: src/test/java/io/vertx/core/VertxTest.java [hunk 3]
```diff
@@ -213,4 +214,42 @@ public void testFinalizeNetClient() throws Exception {
     }
     await();
   }
+
+
+  @Test
+  public void testStickContextFinalization() throws Exception {
+    VertxInternal vertx = (VertxInternal) Vertx.vertx();
```
Comment (jponge) [inline src/test/java/io/vertx/core/VertxTest.java:221]: Not sure you need to cast here.

**Comment by jponge** [comment]
Overall LGTM, and a welcomed change. I'm just really wondering about the opportunity to move `createEventLoopContext` to the public APIs.

**Comment by vietj** [comment]
@jponge for now I'd like to keep it internal, people should deploy verticle in practice to achieve this

**Comment by jponge** [comment]
I agree that users should deploy verticles, I was relating to things like clients where it’d avoid having people cast to VertxInternal.

**Comment by vietj** [comment]
@jponge a cast to VertxInternal is fine, it means you are crossing a boundary. We might take most `*Internal` interfaces at some point and have a `.internal.` package separate from `.impl.`

**Pull request merged**